src/GafferSceneModule/VisibleSetBinding.cpp

```diff
@@ -0,0 +1,114 @@
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "boost/python.hpp"
+
+#include "VisibleSetBinding.h"
+
+#include "GafferScene/ScenePlug.h"
+#include "GafferScene/VisibleSet.h"
+#include "GafferScene/VisibleSetData.h"
+
+#include "IECorePython/RunTimeTypedBinding.h"
+#include "IECorePython/SimpleTypedDataBinding.h"
+
+using namespace boost::python;
+using namespace IECore;
+using namespace GafferScene;
+
+namespace
+{
+
+IECore::PathMatcher &getExpansions( VisibleSet &v )
+{
+	return v.expansions;
+}
+
+void setExpansions( VisibleSet &v, const IECore::PathMatcher &paths )
+{
+	v.expansions = paths;
+}
+
+IECore::PathMatcher &getInclusions( VisibleSet &v )
+{
+	return v.inclusions;
+}
+
+void setInclusions( VisibleSet &v, const IECore::PathMatcher &paths )
+{
+	v.inclusions = paths;
+}
+
+IECore::PathMatcher &getExclusions( VisibleSet &v )
+{
+	return v.exclusions;
+}
+
+void setExclusions( VisibleSet &v, const IECore::PathMatcher &paths )
+{
+	v.exclusions = paths;
+}
+
+IECore::PathMatcher::Result matchWrapper( VisibleSet &v, const ScenePlug::ScenePath &path )
+{
+	return v.match( path );
+}
+
+} // namespace
+
+void GafferSceneModule::bindVisibleSet()
+{
+
+	IECorePython::RunTimeTypedClass<VisibleSetData>()
+		.def( init<>() )
+		.def( init<const VisibleSet &>() )
+		.add_property( "value", make_function( &VisibleSetData::writable, return_internal_reference<1>() ) )
+		.def( "hasBase", &VisibleSetData::hasBase ).staticmethod( "hasBase" )
+	;
+
+	IECorePython::TypedDataFromType<VisibleSetData>();
+
+	class_<VisibleSet>( "VisibleSet" )
+		.def( init<>() )
+		.def( init<const VisibleSet &>() )
+		.def( "match", &matchWrapper )
+		.add_property( "expansions", make_function( &getExpansions, return_internal_reference<>() ), &setExpansions )
+		.add_property( "inclusions", make_function( &getInclusions, return_internal_reference<>() ), &setInclusions )
+		.add_property( "exclusions", make_function( &getExclusions, return_internal_reference<>() ), &setExclusions )
```

Comment:
Is there any reason we can't use `def_readwrite()` here and avoid the additional wrapper functions? It seemed to work when I tried it just now.

Reply:
The only reason for not using `def_readwrite()` here was my complete and utter ignorance of its existence. :) Thanks for enlightening me, it's now being used in 457a78e